src/client/clients/featureFlags/index.ts

```diff
@@ -0,0 +1,82 @@
+import type createClient from "openapi-fetch";
+import {
+  type Context,
+  InMemStorageProvider,
+  type Unleash,
+  initialize,
+} from "unleash-client";
+import type { BaseApiClientParams } from "../../../types/index.js";
+import { FEATURE_FLAGS } from "../../../types/index.js";
+import BaseClient from "../../base-client.js";
+import type { components, paths } from "./schema.js";
+import { waitForConnection } from "./utils.js";
+
+class FeatureFlags extends BaseClient<paths> {
+  declare client: ReturnType<typeof createClient<paths>>;
+  protected serviceName = FEATURE_FLAGS;
+
+  private unleashClient: Unleash | null = null;
+  public isConnected = false;
+
+  constructor({ baseUrl, getTokenFn }: BaseApiClientParams) {
+    super({ baseUrl, getTokenFn });
+    const token = getTokenFn ? (getTokenFn(FEATURE_FLAGS) as string) : "";
+    this.unleashClient = initialize({
+      appName: this.serviceName,
+      url: `${baseUrl}/api`,
+      refreshInterval: 1000,
+      customHeaders: {
+        Authorization: token,
+      },
+      storageProvider: new InMemStorageProvider(),
+    });
+    this.unleashClient.on("error", console.error);
+    this.unleashClient.on("synchronized", () => {
+      this.isConnected = true;
+    });
+  }
+
+  async isFlagEnabled(name: string, context?: Context) {
+    try {
+      // note that there is no waiting if the client is already connected
+      await this.waitForConnection();
+    } catch {
+      return false;
+    }
+    return (
+      (this.isConnected &&
+        this.unleashClient?.isEnabled(name, context, () => false)) ??
+      false
+    );
+  }
+
+  async getFeatureFlags() {
+    return await this.client
+      .GET("/api/admin/projects/{projectId}/features", {
+        params: {
+          path: {
+            projectId: "default",
```

Comment:
shouldn't `projectId` value be dynamic?

```suggestion
  async getFeatureFlags(projected: string) {
    return await this.client
      .GET("/api/admin/projects/{projectId}/features", {
        params: {
          path: {
            projectId,
```

Reply:
Yes. It's not dynamic yet bc it's not a requirement right now, as there can be only 1 project for now. But sure, I can prepare for the future.

Reply:
https://github.com/ogcio/building-blocks-sdk/commit/3e15bf70e14b45bc8690b5b5e53c2e202b6f280a